languages/python/reference/exercise-concepts/reverse-string.md

```diff
@@ -0,0 +1,41 @@
+## Concept extraction for reverse-string
+
+Example implementation:
+```python
+def reverse(text: str) -> str:
+    """Reverse a given string
+    Parameters
+    ----------
+    text : str
+        CAS# for chemical of interest
+    
+    Returns
+    -------
+    [str]
+        the given string in reverse
+    """
+
+    return text[::-1]
+```
+
+### Concepts:
+- **Function**: `def` to create a function in Python
+- **Immutability**: `text` str in Python is [immutable](https://docs.python.org/3/library/stdtypes.html#text-sequence-type-str).
+In this exercise, you return a new string, the old string `text` is not changed.
+- **Return value(s)**: this function return a string by this line: `return text[::-1]`
+- **String [slicing](https://docs.python.org/3/reference/expressions.html#slicings)**: 
+because `str` in python is a `sequence` object, slicing can be used here. Specifically: for syntax `string[start:stop:stride]`:
+  - `start`: 0-index of the start position, `start=0` by default (i.e., not specified) (start from the beginning)
+  - `stop`: 0-index of the stop position, `stop=-1` by default (i.e., not specified) (stop at the end)
+  - `stride`: number of skip step. For example, 
+     ```python
+     >>> string = 'ABCDEF'[::2]
+     >>> print(string)
+     'ACE'
+     ```
+     In this exercise, `stride = -1` means start from the end
```

Comment:
This line is indented too far.

Reply:
I see. Thank you for checking that. I thought I checked the Preview. I should learn markdown better!